### PR TITLE
Incorporate organization in APIM workflows 

### PR DESCRIPTION
**Pull request by Sarangan0219**
### Purpose

Related to https://github.com/wso2/carbon-apimgt/pull/9658

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/OrganizationDTO.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ * /
+ */
+
+package org.wso2.carbon.apimgt.api.model;
+
+public class OrganizationDTO {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/OrganizationDTO.java:24]: Shall we remove this as this is not used?

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java [hunk 2]
```diff
@@ -418,7 +418,7 @@ boolean updateAPIStatus(APIIdentifier apiId, String status,boolean publishToGate
      * @throws APIManagementException If an error occurs while trying to create
      *                                the new version of the API
      */
-    API createNewAPIVersion(String apiId, String newVersion, Boolean defaultVersion)
+    API createNewAPIVersion(String apiId, String newVersion, Boolean defaultVersion, String orgId)
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java:421]:
```suggestion
    API createNewAPIVersion(String apiId, String newVersion, Boolean defaultVersion, String organizationId)
```

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java [hunk 3]
```diff
@@ -1783,6 +1783,6 @@ Mediation updateApiSpecificMediationPolicyContent(String apiId, Mediation mediat
      * @param apiRevisionId API Revision UUID
      * @throws APIManagementException if failed to delete APIRevision
      */
-    void deleteAPIRevision(String apiId, String apiRevisionId) throws APIManagementException;
+    void deleteAPIRevision(String apiId, String apiRevisionId, String organizationId) throws APIManagementException;
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java:1786]: Could not find these methods are called with the new implementation with new organizationId addition?
Are they missed to add to the PR?

Also, I hope these interface changes can be added to the master branch as well.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java [hunk 4]
```diff
@@ -6065,9 +6067,9 @@ private API addTiersToAPI(API api, String requestedTenantDomain) throws APIManag
      * @throws APIManagementException
      */
     @Override
-    public API getLightweightAPIByUUID(String uuid, String requestedTenantDomain) throws APIManagementException {
+    public API getLightweightAPIByUUID(String uuid, String orgId) throws APIManagementException {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java:6070]:
```suggestion
    public API getLightweightAPIByUUID(String uuid, String organizationId) throws APIManagementException {
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java [hunk 5]
```diff
@@ -709,6 +709,7 @@ public static API addAPIWithGeneratedSwaggerDefinition(APIDTO apiDto, String oas
             String apiDefinition = oasParser.generateAPIDefinition(swaggerData);
             apiToAdd.setSwaggerDefinition(apiDefinition);
         }
+        apiToAdd.setOrganizationId(organizationId);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java:712]: shall we do this inside `prepareToCreateAPIByDTO` method itself?

File: components/apimgt/org.wso2.carbon.apimgt.persistence/pom.xml [hunk 6]
```diff
@@ -91,6 +91,27 @@
...
+            <groupId>org.mongodb</groupId>
+            <artifactId>mongodb-driver-core</artifactId>
+            <version>4.1.0</version>
+        </dependency>
+        <dependency>
+            <groupId>net.consensys.cava</groupId>
+            <artifactId>cava-toml</artifactId>
+            <version>0.5.0</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/pom.xml:114]: Looks like these are not related?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java [hunk 7]
```diff
@@ -6065,9 +6067,9 @@ private API addTiersToAPI(API api, String requestedTenantDomain) throws APIManag
...
         try {
-            Organization org = new Organization(requestedTenantDomain);
+            Organization org = new Organization(orgId);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java:6072]: Lets use consistent naming for organization across all places.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java [hunk 8]
```diff
@@ -6210,22 +6212,22 @@ protected void checkAPIVisibilityRestriction(String apiId, String tenantDomain)
     }
     
     @Override
-    public List<Documentation> getAllDocumentation(String uuid, String tenantDomain) throws APIManagementException {
-        checkAPIVisibilityRestriction(uuid, tenantDomain);
-        return super.getAllDocumentation(uuid, tenantDomain);
+    public List<Documentation> getAllDocumentation(String uuid, String orgId) throws APIManagementException {
+        checkAPIVisibilityRestriction(uuid, orgId);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java:6216]: Consistent naming

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIConsumer.java [hunk 9]
```diff
@@ -134,17 +134,9 @@ public API getAPI(APIIdentifier identifier) throws APIManagementException {
...
-        ApiTypeWrapper apiTypeWrapper = super.getAPIorAPIProductByUUID(uuid, requestedTenantDomain);
-        Identifier identifier;
-        if (apiTypeWrapper.isAPIProduct()) {
-            identifier = apiTypeWrapper.getApiProduct().getId();
-        } else {
-            identifier = apiTypeWrapper.getApi().getId();
-        }
-        //TO-DO Commented to since need to work for monodb should be move to registry persistent impl
-//        checkAccessControlPermission(identifier);
+        ApiTypeWrapper apiTypeWrapper = super.getAPIorAPIProductByUUID(uuid, orgId);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIConsumer.java:139]: Is this change necessary?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 10]
```diff
@@ -5724,6 +5724,105 @@ public void makeKeysForwardCompatible(ApiTypeWrapper apiTypeWrapper, String oldV
...
+                ps.setString(1, apiName);
+                ps.setString(2, organizationId);
+                resultSet = ps.executeQuery();
+                while (resultSet.next()) {
+                    versionList.add(resultSet.getString("API_VERSION"));
+                }
+                connection.commit();
+            } catch (SQLException e) {
+                connection.rollback();
+                handleException("Failed to rollback in getting API versions", e);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:5747]:
```suggestion
           APIMgtDBUtil.rollbackConnection(connection, "Failed to rollback get API versions matches API name " + apiName, e);  
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 11]
```diff
@@ -5724,6 +5724,105 @@ public void makeKeysForwardCompatible(ApiTypeWrapper apiTypeWrapper, String oldV
...
+                    APIIdentifier apiId = new APIIdentifier(result.getString("API_PROVIDER"),
+                            result.getString("API_NAME"),
+                            result.getString("API_VERSION"));
+                    API api = new API(apiId);
+                    apis.add(api);
+                }
+                connection.commit();
+            } catch (SQLException e) {
+                connection.rollback();
+                handleException("Failed to rollback in getting APIs", e);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:5782]: Can use the same approach as above

Comment (malinthaprasan) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:5782]: Can remove the second try block and use only the 1st one?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 12]
```diff
@@ -5724,6 +5724,105 @@ public void makeKeysForwardCompatible(ApiTypeWrapper apiTypeWrapper, String oldV
...
+            } finally {
+                APIMgtDBUtil.setAutoCommit(connection, initialAutoCommit);
+            }
+        } catch (SQLException e) {
+            handleException("Error occurred while fetching organization ID", e);
+        }
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:5820]: Can use a single try block?

Comment (malinthaprasan) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:5820]: Please check and fix other places too

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ExportUtils.java [hunk 13]
```diff
@@ -778,10 +779,13 @@ public static void addAPIMetaInformationToArchive(String archivePath, APIDTO api
                                 exportFormat, ImportExportConstants.GRAPHQL_COMPLEXITY, graphQLQueryComplexityInfoDTO);
                     }
                 }
+                API api = APIMappingUtil.fromDTOtoAPI(apiDtoToReturn, apiDtoToReturn.getProvider());
+                //TODO-ORG
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ExportUtils.java:783]: What are the things remaining to do here?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java [hunk 14]
```diff
@@ -69,17 +69,7 @@
...
-import org.wso2.carbon.apimgt.api.model.APIProduct;
-import org.wso2.carbon.apimgt.api.model.APIProductIdentifier;
-import org.wso2.carbon.apimgt.api.model.APIResourceMediationPolicy;
-import org.wso2.carbon.apimgt.api.model.APIRevision;
-import org.wso2.carbon.apimgt.api.model.APIRevisionDeployment;
-import org.wso2.carbon.apimgt.api.model.APIStateChangeResponse;
-import org.wso2.carbon.apimgt.api.model.APIStore;
-import org.wso2.carbon.apimgt.api.model.Documentation;
-import org.wso2.carbon.apimgt.api.model.DocumentationContent;
+import org.wso2.carbon.apimgt.api.model.*;
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java:72]: Let's use individual imports.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java [hunk 15]
```diff
@@ -3137,11 +3061,12 @@ public Response importWSDLDefinition(InputStream fileInputStream, Attachment fil
...
             additionalPropertiesAPI.setType(APIDTO.TypeEnum.fromValue(implementationType));
-            API apiToAdd = PublisherCommonUtils
-                    .prepareToCreateAPIByDTO(additionalPropertiesAPI, RestApiCommonUtil.getLoggedInUserProvider(),
-                            username);
+            API apiToAdd = PublisherCommonUtils.prepareToCreateAPIByDTO(additionalPropertiesAPI,
+                    RestApiCommonUtil.getLoggedInUserProvider(), username, organizationId);
+            apiToAdd.setOrganizationId(organizationId);
             apiToAdd.setWsdlUrl(url);
             API createdApi = null;
+            apiToAdd.setOrganizationId(organizationId);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java:3069]: Duplicated?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java [hunk 16]
```diff
@@ -3294,7 +3219,8 @@ private API importSOAPToRESTAPI(InputStream fileInputStream, Attachment fileDeta
                     throw new APIManagementException(ExceptionCodes.UNSUPPORTED_WSDL_FILE_EXTENSION);
                 }
             }
-            String updatedSwagger = updateSwagger(createdApi.getUUID(), swaggerStr);
+            //TODO-ORG
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java:3222]: SOAP to REST is not required for the initial implementation.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java [hunk 17]
```diff
@@ -3455,10 +3376,8 @@ public Response createNewAPIVersion(String newVersion, String apiId, Boolean def
                         apiId));
             }
             APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
-            String tenantDomain = RestApiCommonUtil.getLoggedInUserTenantDomain();
-
-            API versionedAPI = apiProvider.createNewAPIVersion(apiId, newVersion, defaultVersion);
-
+            //TODO- ORG
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java:3379]: Can we remove the Todo?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 18]
```diff
@@ -105,57 +105,58 @@ info:
     url: http://www.apache.org/licenses/LICENSE-2.0.html
   version: v2
 servers:
-- url: https://apis.wso2.com/api/am/publisher/v2
+  - url: https://apis.wso2.com/api/am/publisher/v2
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:108]: I would suggest not to do non-related formatting changes to this with this PR to publisher-api.yaml. If we are doing this let's do it in a separate PR and do the same in the master branch too.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java [hunk 19]
```diff
@@ -27,16 +27,7 @@
...
-import org.wso2.carbon.apimgt.api.model.APIIdentifier;
-import org.wso2.carbon.apimgt.api.model.APIRating;
-import org.wso2.carbon.apimgt.api.model.ApiTypeWrapper;
-import org.wso2.carbon.apimgt.api.model.Comment;
-import org.wso2.carbon.apimgt.api.model.Documentation;
-import org.wso2.carbon.apimgt.api.model.DocumentationContent;
-import org.wso2.carbon.apimgt.api.model.Identifier;
-import org.wso2.carbon.apimgt.api.model.ResourceFile;
-import org.wso2.carbon.apimgt.api.model.Tier;
+import org.wso2.carbon.apimgt.api.model.*;
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java:30]: Let's use individual imports.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/devportal-api.yaml [hunk 20]
```diff
@@ -106,15 +106,15 @@ info:
     url: http://www.apache.org/licenses/LICENSE-2.0.html
   version: v2
 servers:
-- url: https://apis.wso2.com/api/am/devportal/v2
+  - url: https://apis.wso2.com/api/am/devportal/v2
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/devportal-api.yaml:109]: Let's do not format the full file?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/devportal-api.yaml [hunk 21]
```diff
@@ -1206,14 +1225,15 @@ paths:
...
-      - Applications
+        - Applications
       summary: |
         Get Details of an Application
       description: |
         This operation can be used to retrieve details of an individual application specifying the application id in the URI.
       parameters:
-      - $ref: '#/components/parameters/applicationId'
-      - $ref: '#/components/parameters/If-None-Match'
+        - $ref: '#/components/parameters/organizationId'
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/devportal-api.yaml:1234]: I think we don't need organizationId here?
Can you please check and remove from all the unwanted places?

**Review by malinthaprasan — COMMENTED** [review]
Do not commit org.wso2.carbon.apimgt.core.feature/src/main/resources/repository/database/WSO2AM_DB.mv.db

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java (hunk 2 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java:421]: As we discussed offline, **orgId**  will be used in interfaces. On the implementation side, we use **tenantDomain** and **organizationId** .

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java (hunk 3 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java:1786]: Yes. These changes will be added to the master

File: components/apimgt/org.wso2.carbon.apimgt.persistence/pom.xml (hunk 6 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/pom.xml:114]: This will be removed in this PR. WIll be Added through MongoDB implementation

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIConsumer.java (hunk 9 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIConsumer.java:139]: That identifier object is not used anywhere. Thus removed it from the implementation.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ExportUtils.java (hunk 13 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ExportUtils.java:783]: That has been already implemented. Removed that comment

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/OrganizationDTO.java (hunk 1 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/OrganizationDTO.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/cc8517b20f7dbec93569cf9ab02809b6b9148cbe

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java (hunk 4 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/cc8517b20f7dbec93569cf9ab02809b6b9148cbe

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java (hunk 7 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/cc8517b20f7dbec93569cf9ab02809b6b9148cbe

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java (hunk 8 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/cc8517b20f7dbec93569cf9ab02809b6b9148cbe

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java (hunk 5 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/cc8517b20f7dbec93569cf9ab02809b6b9148cbe

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ExportUtils.java (hunk 13 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ExportUtils.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/cc8517b20f7dbec93569cf9ab02809b6b9148cbe

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java (hunk 14 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/cc8517b20f7dbec93569cf9ab02809b6b9148cbe

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java (hunk 15 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/cc8517b20f7dbec93569cf9ab02809b6b9148cbe

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java (hunk 16 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/cc8517b20f7dbec93569cf9ab02809b6b9148cbe

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java (hunk 17 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/cc8517b20f7dbec93569cf9ab02809b6b9148cbe

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java (hunk 19 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/cc8517b20f7dbec93569cf9ab02809b6b9148cbe

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java (hunk 11 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: The inner try-catch block will handle all the transaction-related operations and the outer try block will handle connection related operations. When removing this nested try block, we can't differentiate the error stack traces in these two operations.  

WDYT ?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java (hunk 12 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:5819]: The inner try-catch block will handle all the transaction-related operations and the outer try block will handle connection related operations. When removing this nested try block, we can't differentiate the error stack traces in these two operations.  

WDYT ?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java (hunk 10 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/9869328646edc095172b37c6bb8f1b2434656efa

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/devportal-api.yaml (hunk 21 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/devportal-api.yaml]: Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/0674510dd7d56bfe90e23dcc59886ea7b77cf741

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml (hunk 18 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:108]: As discussed offline, we will send a PR to the master branch addressing all the formatting changes.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/devportal-api.yaml (hunk 20 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/devportal-api.yaml:109]: As discussed offline, we will send a PR to the master branch addressing all the formatting changes.

**Review by malinthaprasan — APPROVED** [review]
